aspects/aspects.go

```diff
@@ -123,22 +124,15 @@ func NewAspectDirectory(name string, aspects map[string]interface{}, dataBag Dat
 				return nil, errors.New(`cannot create aspect pattern without a "name" field`)
 			}
 
-			// TODO: either
-			// * Validate that a path isn't a subset of another
-			//   (possibly somewhere else).  Otherwise, we can
-			//   write a user value in a subkey of a path (that
-			//   should be map).
-			// * Our schema should be able to provide
-			//   allowed/expected types given a path; these should
-			//   guide and take precedence resolving conflicts
-			//   between data in the data bags or written E.g
-			//   possibly return null or empty object if at a path
-			//   were the schema expects an object there is scalar?
 			path, ok := aspectPattern["path"]
 			if !ok || path == "" {
 				return nil, errors.New(`cannot create aspect pattern without a "path" field`)
 			}
 
+			if err := validateNamePathPair(name, path); err != nil {
+				return nil, err
+			}
+
 			accPattern, err := newAccessPattern(name, path, aspectPattern["access"])
 			if err != nil {
 				return nil, err
```

Comment:
it feels like the whole loop should be wrapped in such a way that we can produce errors that mention the aspect Name?